lib/shared/types.js

```diff
@@ -125,7 +125,7 @@ module.exports = {};
  * @property {Record<string,string>} [messages] The messages the rule reports.
  * @property {string[]} [replacedBy] The IDs of the alternative rules.
  * @property {Array|Object} schema The option schema of the rule.
- * @property {"problem"|"suggestion"|"layout"} type The rule type.
+ * @property {"directive"|"problem"|"suggestion"|"layout"} type The rule type.
```

Comment:
I think this is a typedef for `meta` properties in rule objects, so `rule.meta.type` cannot be `"directive"`.